### PR TITLE
Refactored "set settings"

### PR DESCRIPTION
**Pull request by JanPetr**
- Improved logging
- Extracted to standalone class
- "Data" helper is not responsible for "set settings" anymore

### CODE REVIEW

File: Helper/Entity/ProductHelper.php [hunk 1]
```diff
@@ -301,8 +301,10 @@ public function setSettings($indexName, $indexNameTmp, $storeId, $saveToTmpIndic
         $indexSettings = $transport->getData();
 
         $this->algoliaHelper->setSettings($indexName, $indexSettings, false, true);
+        $this->logger->log('Settings: ' . json_encode($indexSettings));
```
Comment (damcou) [inline Helper/Entity/ProductHelper.php]: How about adding the index name here ?

File: Test/Integration/ConfigTest.php [hunk 2]
```diff
@@ -2,18 +2,16 @@
...
 {
     public function testFacets()
     {
-        $facets = $this->configHelper->getFacets();
-
-        /** @var Data $helper */
-        $helper = $this->getObjectManager()->create('Algolia\AlgoliaSearch\Helper\Data');
-        $helper->saveConfigurationToAlgolia(1);
+        /** @var IndicesConfigurator $indicesConfiurator */
+        $indicesConfiurator = $this->getObjectManager()->create(IndicesConfigurator::class);
```
Comment (damcou) [inline Test/Integration/ConfigTest.php]: Missing "g" :D

File: Test/Integration/ConfigTest.php [hunk 3]
```diff
@@ -24,9 +22,9 @@ public function testFacets()
 
     public function testQueryRules()
     {
-        /** @var Data $helper */
-        $helper = $this->getObjectManager()->create('Algolia\AlgoliaSearch\Helper\Data');
-        $helper->saveConfigurationToAlgolia(1);
+        /** @var IndicesConfigurator $indicesConfiurator */
+        $indicesConfiurator = $this->getObjectManager()->create(IndicesConfigurator::class);
```
Comment (damcou) [inline Test/Integration/ConfigTest.php]: Same everywhere after :)

File: Helper/Entity/ProductHelper.php (hunk 1 above)
Comment (JanPetr) [reply Helper/Entity/ProductHelper.php]: The index name is already logged here: https://github.com/algolia/algoliasearch-magento-2/pull/675/files#diff-55d7adf73efa24c6e154bba2a58b6d58R201

File: Test/Integration/ConfigTest.php (hunk 2 above)
Comment (JanPetr) [reply Test/Integration/ConfigTest.php]: Fixed, good catch :))

**Pull request merged**